src/client/src/components/ListControls/ListControls.jsx

```diff
@@ -1,41 +1,48 @@
-import React, { useState, useEffect } from "react";
-import PropTypes from "prop-types";
-import { Input } from "../Input/Input";
-import { Button } from "../Button/Button";
-import { useItem } from "../../hooks/useItem";
+import React, { useState, useCallback } from "react";
+import { connect } from "react-redux";
+import { bindActionCreators } from "redux";
+import { TextFieldComponent } from "../TextField/TextField";
+import { ButtonComponent } from "../Button/Button";
+import { getItemsList } from "../../selectors/itemsEntitiesSelectors";
+import { addItemAction } from "../../actions/itemsEntitiesActions";
 import "./listControls.css";
 
-export const ListControls = ({ renderItems }) => {
-  const { addItem } = useItem();
+const ListControls = ({ addItemAction }) => {
   const [inputValue, setInputValue] = useState("");
 
-  useEffect(() => {
-    renderItems();
-  }, [inputValue, renderItems]);
+  const onInputChange = useCallback(
+    (e) => {
+      setInputValue(e);
+    },
+    [setInputValue]
+  );
 
-  const handleItem = async () => {
-    await addItem(inputValue);
-    await renderItems();
+  const handleItem = useCallback(async () => {
+    addItemAction({ itemName: inputValue });
 
     setInputValue("");
-  };
+  }, [addItemAction, inputValue]);
 
   return (
     <div className="list-controls">
-      <Input
+      <TextFieldComponent
         placeholder={"Add your new todo"}
         value={inputValue}
-        onChange={(e) => setInputValue(e.target.value)}
+        onChange={onInputChange}
       />
-      <Button label={"+"} onClick={handleItem}></Button>
+      <ButtonComponent label={"+"} onClick={handleItem}></ButtonComponent>
     </div>
   );
 };
 
-ListControls.propTypes = {
-  renderItems: PropTypes.func,
+const mapStateToProps = (state, ownProps) => {
+  const itemsList = getItemsList(state);
```

Comment:
move mapStateToProps, mapDispatchToProps and the connect code to a separate connector file.